### DISCUSSION TITLE
Questions about contributing

### DISCUSSION THREAD

**Discussion by fkaute** [ComplianceAsCode/content#11182] [General]
Hi all,

I'm working on an ansible role for hardening Ubuntu 22.04 server with the help of all these ansible code snippets from the SSG reports. I also do check the cod with ansible-lint. When it comes to lint I found many small errors like indents or block/name/when/tags order or FQCN.

I also have 1-2 code snippets where ansible-lint required some deeper changes like moving from code that uses command module with systemctl commands towards systemd module  using.

I wonder wether I should contribute those changes because thats how opensource works or if these code snippets are more like a hint for people to find their own solution like I just did.

Best regards

**Comment by SadeeshaJayaweera** [comment]
Contributing to open source projects is a great way to give back to the community and help improve the quality of code for everyone. If you've identified issues with code snippets in an open source project and have suggestions for improvement, it's generally a good practice to contribute those changes, especially if they can make the project more robust, maintainable, and aligned with best practices.

Here are some considerations:

1. **Community Benefit:** Contributing your improvements can benefit the entire community. Others may encounter the same issues or challenges and could benefit from your solutions.

2. **Maintainer's Perspective:** Open source maintainers often appreciate contributions that enhance their project. They may not have had the time or resources to make these changes themselves.

3. **Collaboration:** Open source is built on collaboration. By contributing, you're participating in a collaborative effort to make the project better.

4. **Documentation:** If the code snippets are intended as examples for others, improving them can also enhance the documentation and provide clearer guidance to users.

5. **Acknowledgment:** When you contribute to open source, you can get acknowledgment for your work, which can be valuable for your own profile and reputation.

6. **License Compatibility:** Ensure that your contributions comply with the project's licensing terms.

Before contributing, here's what you can do:

- **Understand the Project**: Familiarize yourself with the project's guidelines, coding standards, and contribution process. Check if there are any existing issues or discussions related to the code snippets you want to improve.

- **Create Clear Proposals**: Clearly explain the issues you've found and the proposed changes. This can include code examples, explanations, and references to best practices or relevant documentation.

- **Engage with the Community**: If possible, engage with the project's community through discussions, mailing lists, or forums to gather feedback and build consensus on your proposed changes.

- **Follow Contribution Guidelines**: Ensure that you follow the project's contribution guidelines, including code formatting, documentation, and testing requirements.

- **Submit a Pull Request**: If the project uses a version control system like Git, submit a pull request with your proposed changes. This makes it easier for maintainers and reviewers to evaluate and merge your contributions.

Ultimately, whether or not to contribute depends on your willingness to invest time and effort in improving the project. If you believe your changes can benefit the project and its users, it's a valuable contribution to make.
